### PR TITLE
SAK-50266 Profile replace HibernateTemplate with modern Session pattern

### PR DESCRIPTION
**Pull request by ern**
https://sakaiproject.atlassian.net/browse/SAK-50266

### CODE REVIEW

**Review by ottenhoff — APPROVED** [review]

File: profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java [hunk 1]
```diff
@@ -34,76 +32,40 @@
...
-				privacy = (ProfilePrivacy)cache.get(userId);
-				if(privacy != null) {
-				  return(privacy);
-				}
-				// This means that the cache has expired. evict the key from the cache
-				log.debug("Privacy cache appears to have expired for " + userId);
-				this.cacheManager.evictFromCache(this.cache, userId);
-			}
+		if (StringUtils.isBlank(userId)) {
+			throw new IllegalArgumentException("Invalid userid " + userId);
```
Comment (jesusmmp) [inline profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java:47]: throw new IllegalArgumentException("Invalid userid {}, userId);

File: profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java [hunk 2]
```diff
@@ -113,25 +75,14 @@ public boolean savePrivacyRecord(ProfilePrivacy privacy) {
...
 		
-		//save
-		if(dao.updatePrivacyRecord(privacy)) {
-			log.info("Saved privacy record for user: " + privacy.getUserUuid()); 
-			
-			//update cache
-			log.debug("Updated privacy record in cache for: " + privacy.getUserUuid());
-			cache.put(privacy.getUserUuid(), privacy);
-			
+		if (dao.updatePrivacyRecord(privacy)) {
```
Comment (jesusmmp) [inline profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java:78]: Could be use the return directly?
return dao.updatePrivacyRecord(privacy)

File: profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java [hunk 3]
```diff
@@ -143,18 +94,16 @@ public boolean isActionAllowed(final String userX, final String userY, final Pri
...
     	
     	boolean isConnected = connectionsLogic.isUserXFriendOfUserY(userX, userY);
 		
     	boolean result=false;
     	
-    	if(log.isDebugEnabled()){
-	    	log.debug("ProfilePrivacyLogic.isActionAllowed. userX: " + userX + ", userY: " + userY + ", type: " + type);  
-    	}
-    	
+		log.debug("userX: {}, userY: {}, type: {}", userX, userY, type);
```
Comment (jesusmmp) [inline profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java:105]: Spacing

File: profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java (hunk 1 above)
Comment (ottenhoff) [reply profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java:47]: i don't think you can do that in a throw, can you?

File: profile2/impl/src/java/org/sakaiproject/profile2/dao/impl/ProfileDaoImpl.java [hunk 4]
```diff
@@ -45,1155 +35,649 @@
...
+            log.warn("confirm friend request failed, {}", e.toString());
+        }
+        return false;
+    }
+
+    @Override
+    public boolean removeConnection(final ProfileFriend profileFriend) {
+        try {
+            Session session = getSession();
+            session.remove(session.merge(profileFriend));
```
Comment (jesusmmp) [inline profile2/impl/src/java/org/sakaiproject/profile2/dao/impl/ProfileDaoImpl.java:178]: session.remove is ok?

File: profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfileKudosLogicImpl.java [hunk 5]
```diff
@@ -95,21 +64,9 @@ public boolean updateKudos(String userUuid, int score, BigDecimal percentage) {
 		k.setDateAdded(new Date());
 		
 		if(dao.updateKudos(k)){
```
Comment (jesusmmp) [inline profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfileKudosLogicImpl.java:66]: return dao.updateKudos(k)?

File: profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java [hunk 6]
```diff
@@ -34,76 +32,40 @@
...
-		if(privacy != null) {
-			log.debug("Adding privacy record to cache for: " + userId);
-			cache.put(userId, privacy);
 		}
-		
-		//if still null, we can't do much except log an error and wait for an NPE.
-		if(privacy == null) {
-			log.error("Couldn't retrieve or create a privacy record for user: " + userId + " This is an error and you need to fix your installation.");
+
+		if (privacy == null) {
```
Comment (jesusmmp) [inline profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java:59]: This dao.addNewPrivacyRecord(getDefaultPrivacyRecord(userId)); could return null?

File: profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java (hunk 1 above)
Comment (jesusmmp) [reply profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java:47]: You are right! :)

**Comment by jesusmmp** [comment]
Some little comments. It looks so good! Awesome job

File: profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java (hunk 6 above)
Comment (ern) [reply profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java:59]: correct that is how it was previously

File: profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java (hunk 2 above)
Comment (ern) [reply profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java:78]: updated

**Review by jesusmmp — APPROVED** [review]

File: profile2/impl/src/java/org/sakaiproject/profile2/dao/impl/ProfileDaoImpl.java (hunk 4 above)
Comment (ern) [reply profile2/impl/src/java/org/sakaiproject/profile2/dao/impl/ProfileDaoImpl.java:178]: yes, it is the jpa version of delete
https://docs.jboss.org/hibernate/orm/5.4/userguide/html_single/Hibernate_User_Guide.html#pc-remove

**Comment by ern** [comment]
thx Jesus for the review, much appreciated

**Pull request merged**